### PR TITLE
Updated documentation in this repo

### PR DESCRIPTION
**Pull request by KarishmaBothara**
With this PR, have modified all the doc examples and tested it on local... I could not successful run the code to upgrade runtime, it gives me error as - 
```
Upgrading from 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY, 1405969 bytes
2020-11-23 13:38:16        RPC-CORE: submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1010: Invalid Transaction: Transaction would exhaust the block limits
2020-11-23 13:38:16             DRR: Error: 1010: Invalid Transaction: Transaction would exhaust the block limits
```
However, the same code is being used by polkadot in their example here https://polkadot.js.org/docs/api/examples/promise/upgrade-chain/

I managed to fix script to generate documentation i.e scripts/MetadataMd.ts and now docs/cennznet has all the relevant information fetched from the metadata.

### CODE REVIEW

File: docs/cennznet/constants.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
+## Constants
```
Comment (KarishmaBothara) [inline docs/cennznet/constants.md:1]: Auto generated constants from metadata

File: docs/cennznet/errors.md [hunk 2]
```diff
@@ -0,0 +1,492 @@
+## Errors
+
```
Comment (KarishmaBothara) [inline docs/cennznet/errors.md:2]: Auto generated errors from metadata

File: docs/cennznet/events.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
+## Events
+
```
Comment (KarishmaBothara) [inline docs/cennznet/events.md:2]: Auto generated events from metadata

File: docs/cennznet/extrinsics.md [hunk 4]
```diff
@@ -0,0 +1,1975 @@
+## Extrinsics
+
```
Comment (KarishmaBothara) [inline docs/cennznet/extrinsics.md:2]: Auto generated extrinsic from metadata

File: docs/cennznet/rpc.md [hunk 5]
```diff
@@ -0,0 +1,370 @@
+## JSON-RPC
+
```
Comment (KarishmaBothara) [inline docs/cennznet/rpc.md:2]: Auto generated rpc (from definitions.ts)
Also includes genericAssets and Cennzx RPC calls

File: docs/cennznet/storage.md [hunk 6]
```diff
@@ -0,0 +1,800 @@
+## Storage
```
Comment (KarishmaBothara) [inline docs/cennznet/storage.md:1]: Auto generated storage from metadata

File: docs/examples/promise/04_listen_to_multiple_balances_change/index.js [hunk 7]
```diff
@@ -22,7 +22,7 @@ async function main () {
 
   // Subscribe and listen to several balance changes
   api.query.genericAsset.freeBalance.multi([[CENNZ, ALICE], [CENNZ, BOB]], (balances) => {
-    console.log('Change detected, new balances: ', balances);
+    console.log('Change detected, new balances: ', balances.toString());
```
Comment (KarishmaBothara) [inline docs/examples/promise/04_listen_to_multiple_balances_change/index.js:25]: Change detected, new balances:  [ <BN: 33b2e3c9fd0803ce7ff6f55>, <BN: 33b2e3c9fd0803ce80090ab> ]
Change detected, new balances:  [ <BN: 33b2e3c9fd0803ce7ff3f1c>, <BN: 33b2e3c9fd0803ce800c0e4> ]

changed the appearance to

Change detected, new balances:  999999999999999999999950620,1000000000000000000000049380
Change detected, new balances:  999999999999999999999938275,1000000000000000000000061725

File: docs/GET_STARTED.md [hunk 8]
```diff
@@ -95,9 +80,8 @@ const hash = await extrinsic.signAndSend(alice);
 ## FAQ
 > What is the difference between @cennznet/types and @polkadot/types?
 
-We have custom types needed to interact with our specific modules such as Doughnut, CENNZxSpot, Generic Asset, Attestation and etc which you can only find in @cennznet/types. 
+We have custom types needed to interact with our specific modules such as CENNZxSpot, Generic Asset, Attestation and etc which you can only find in @cennznet/types.
```
Comment (holygits) [inline docs/GET_STARTED.md:83]:
```suggestion
This project has custom types needed to interact with CENNZnet specific modules such as CENNZX, Generic Asset, Attestation etc. see @cennznet/types.
There are also CENNZnet only RPC methods and extrinsic customizations provided by @cennznet/api
```

File: docs/examples/promise/03_listen_to_balance_change/index.js [hunk 9]
```diff
@@ -24,7 +24,7 @@ async function main () {
   // Here we subscribe to any balance changes and update the on-screen value
   api.query.genericAsset.freeBalance(CENNZ, Alice, (current) => {
     // Calculate the delta
-    const change = current.sub(previous);
+    const change = previous.sub(current);
```
Comment (holygits) [inline docs/examples/promise/03_listen_to_balance_change/index.js:27]: why this change 😅

File: docs/examples/promise/09_make_transfer_with_allowed_block_permissions_only/index.js [hunk 10]
```diff
@@ -2,10 +2,11 @@
 /* eslint-disable @typescript-eslint/unbound-method */
 /* eslint-disable @typescript-eslint/no-var-requires */
 // Import the API, Keyring and some utility functions
-const { Api } = require('@cennznet/api');
-const testKeyring = require('@polkadot/keyring/testing');
-const { createType } = require('@cennznet/types');
 
+// const ExtrinsicEra = require( '@polkadot/types/primitive/Extrinsic/ExtrinsicEra');
```
Comment (holygits) [inline docs/examples/promise/09_make_transfer_with_allowed_block_permissions_only/index.js]: is this still needed?

File: docs/examples/promise/12_upgrade_chain/index.js [hunk 11]
```diff
@@ -5,8 +5,7 @@
 const { Api } = require('@cennznet/api');
```
Comment (holygits) [inline docs/examples/promise/12_upgrade_chain/index.js:6]: ` : api.tx.consensus.setCode(`0x${code}`); // For previous versions` this code is long removed.

I don't think this example has much value to ordinary users so would prefer to remove it

File: docs/examples/rx/09_upgrade_chain/index.js [hunk 12]
```diff
@@ -1,7 +1,7 @@
 // Import the API & Provider and some utility functions
 const { ApiRx } = require('@cennznet/api');
 // import the test keyring (already has dev keys for Alice, Bob, Charlie, Eve & Ferdie)
-const testKeyring = require('@polkadot/keyring/testing');
+const { Keyring } = require('@polkadot/keyring');
```
Comment (holygits) [inline docs/examples/rx/09_upgrade_chain/index.js:4]: same here not useful to most developers so would prefer to remove

File: docs/examples/promise/09_make_transfer_with_allowed_block_permissions_only/index.js (hunk 10 above)
Comment (KarishmaBothara) [reply docs/examples/promise/09_make_transfer_with_allowed_block_permissions_only/index.js]: no.. I already removed this..in my clean up commit

File: docs/examples/promise/03_listen_to_balance_change/index.js (hunk 9 above)
Comment (KarishmaBothara) [reply docs/examples/promise/03_listen_to_balance_change/index.js:27]: I can leave it what it was before. Actually.. I tried to transfer balance from Alice to Bob.. so
previous balance > current balance and it would show in negative..
but it can also be the other way around, when someone is trying to send some balance to Alice..
I will revert this change

File: docs/examples/promise/12_upgrade_chain/index.js (hunk 11 above)
Comment (KarishmaBothara) [reply docs/examples/promise/12_upgrade_chain/index.js:6]: ya.. will remove it..

File: scripts/MetadataMd.ts [hunk 13]
```diff
@@ -303,14 +302,14 @@ function writeFile (name: string, ...chunks: any[]): void {
...
...0000000000000000000004bc20416363756d756c61746564207472616e73616374696f6e206665657320666f7220726577617264207061796f7574605472616e73616374696f6e466565506f74486973746f727901005856656344657175653c42616c616e63654f663c543e3e040004dc20486973746f72696320616363756d756c61746564207472616e73616374696f6e2066656573206f6e20726577617264207061796f75740108487365745f696e666c6174696f6e5f7261746508246e756d657261746f720c6936342c64656e6f6d696e61746f720c693634045501205365742074686520706572207061796f757420696e666c6174696f6e20726174652028606e756d657261746f7260202f206064656e6f6d696e61746f72602920286974206d6179206265206e6567617469766529647365745f646576656c6f706d656e745f66756e645f74616b6504406e65775f74616b655f70657263656e740c75333204c4205365742074686520646576656c6f706d656e742066756e642074616b6520252c2063617070656420617420313030252e0104305265776172645061796f7574081c42616c616e63651c42616c616e636504b4204120726577617264207061796f75742068617070656e656420287061796f75742c2072656d61696e6465722900001e041c40436865636b5370656356657273696f6e38436865636b547856657273696f6e30436865636b47656e6573697338436865636b4d6f7274616c69747928436865636b4e6f6e63652c436865636b576569676874604368617267655472616e73616374696f6e5061796d656e74';
```
Comment (alexsednz) [inline scripts/MetadataMd.ts:305]: Why meta has copied here directly?

File: docs/examples/promise/03_listen_to_balance_change/index.js (hunk 9 above)
Comment (KarishmaBothara) [reply docs/examples/promise/03_listen_to_balance_change/index.js]: updated this to check if balance is sent or transferred from Alice's account..

File: scripts/MetadataMd.ts (hunk 13 above)
Comment (KarishmaBothara) [reply scripts/MetadataMd.ts]: I had in mind to use it from static meta data file.. have used it here..

File: docs/examples/rx/03_listen_to_balance_change/index.js [hunk 14]
```diff
@@ -30,7 +30,12 @@ async function main () {
         return;
       }
 
-      const change = balance[1].sub(balance[0]);
+      let change;
```
Comment (holygits) [inline docs/examples/rx/03_listen_to_balance_change/index.js:33]: there's a function for this kind of thing: `Math.abs()`

**Review by holygits — APPROVED** [review]
thanks for this!

File: docs/examples/rx/03_listen_to_balance_change/index.js (hunk 14 above)
Comment (KarishmaBothara) [reply docs/examples/rx/03_listen_to_balance_change/index.js:33]: I will check..if balance type has this.. cause all these calculation are being performed on balance type.. If it were a number then I could use Math.abs()
